.github/workflows/ci-e2e-badger.yaml

```diff
@@ -1,14 +1,10 @@
 name: CIT Badger
 
 on:
-  push:
-    branches: [main]
-
-  pull_request:
-    branches: [main]
+  workflow_call:
 
 concurrency:
-  group: ${{ github.workflow }}-${{ (github.event.pull_request && github.event.pull_request.number) || github.ref || github.run_id }}
+  group: cit-badger-${{ github.workflow }}-${{ (github.event.pull_request && github.event.pull_request.number) || github.ref || github.run_id }}
```

Comment:
Q: instead of hardcoding strings like `cit-badger`, could we use some standard variables like workflow-name (which we already use) and job name?

Reply:
${{ github.workflow }}-${{ github.job }}-${{ (github.event.pull_request && github.event.pull_request.number) || github.ref || github.run_id }} you mean like this?

Reply:
Should this be included in the combined CIT workflow as well?

Reply:
Will this be the same for the combined CIT workflow as well?

Reply:
good question - what's the granularity of the concurrency settings? Do we need this setting on the combined workflow if individual jobs can cancel themselves as needed?

Reply:
We might not need concurrency at the combined workflow level unless there are scenarios where concurrent triggers could occur. 
 1. A push to the `main` branch and a PR to `main` happening simultaneously.
 2. Manual trigger while a push event is in progress.
 Should we proceed with adding concurrency here, or leave it out for now?

Reply:
In those 2 scenarios what would concurrency on combine workflow achieve that won't be covered by the job-level settings? All we want is to cancel a job for a PR or a branch if a new version of the PR or branch is available, so as to not waste the runner resources.

Reply:
I was thinking along the same lines—are we agreeing that no concurrency at the combined workflow level is the approach here, relying instead on job-level concurrency to handle redundant jobs efficiently

Reply:
> `${{ github.workflow }}-${{ github.job }}-${{ (github.event.pull_request && github.event.pull_request.number) || github.ref || github.run_id }}`

can you try & test this? If you already have a setup in a private repo, the test would be to start a workflow, open some of the runs in a tab, then push a new commit and observe those runs to be cancelled. Then repeat the same for a branch push instead of a PR.

Reply:
Tested both scenarios as discussed, with the following observations:

Scenario 1: PR with Multiple Commits
Initially, used concurrency at the combined level:
combined-cit-${{ github.workflow }}-${{ (github.event.pull_request && github.event.pull_request.number) || github.ref || github.run_id }}
along with reference-level concurrency:
${{ github.workflow }}-${{ github.job }}-${{ (github.event.pull_request && github.event.pull_request.number) || github.ref || github.run_id }}
This setup caused issues—some referenced workflows in the combined workflow were executed while others were cancelled, even on creating a simple PR.
Removed concurrency at the combined level and retained reference-level concurrency as:
cit-badger-${{ github.workflow }}-${{ github.job }}-${{ (github.event.pull_request && github.event.pull_request.number) || github.ref || github.run_id }}
With this, the workflows behaved as expected, and previous jobs were cancelled when new commits were pushed to the PR.
Scenario 2: Branch Push
Tested with no concurrency at the combined level and only the reference-level concurrency:
cit-badger-${{ github.workflow }}-${{ github.job }}-${{ (github.event.pull_request && github.event.pull_request.number) || github.ref || github.run_id }}
The previous jobs were successfully cancelled when new commits were pushed to the branch.
Observations
Keeping or removing concurrency at the combined level didn’t change the outcome, as long as the reference-level concurrency was properly configured.
For the tests to work reliably, the concurrency group needed a prefix like cit-badger or similar. Without this prefix, referenced workflows in the combined workflow might execute inconsistently (some running and others cancelling) even when just creating a PR.
Should we finalize this setup with no concurrency at the combined level and the prefix-based approach at the reference level? wdyt?